### PR TITLE
vscode-extensions: init vim, github, theme-onelight

### PR DESCRIPTION
**Pull request by garbas**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/misc/vscode-extensions/default.nix [hunk 1]
```diff
@@ -28,4 +28,11 @@ rec {
   ms-python.python = callPackage ./python {};
 
   WakaTime.vscode-wakatime = callPackage ./wakatime {};
+
+  vscodevim.vim = callPackage ./vim {};
+
+  github.vscode-pull-request-github = callPackage ./github {};
+
+  akamud.vscode-theme-onelight = callPackage ./theme-onelight {};
+
```
Comment (veprbl) [inline pkgs/misc/vscode-extensions/default.nix]: Nitpick: can we do order alphabetic by namespace name? :)

File: pkgs/misc/vscode-extensions/default.nix [hunk 2]
```diff
@@ -28,4 +28,11 @@ rec {
   ms-python.python = callPackage ./python {};
 
   WakaTime.vscode-wakatime = callPackage ./wakatime {};
+
+  vscodevim.vim = callPackage ./vim {};
+
+  github.vscode-pull-request-github = callPackage ./github {};
```
Comment (veprbl) [inline pkgs/misc/vscode-extensions/default.nix]:
```suggestion
  github.vscode-pull-request-github = callPackage ./pull-request-github {};
```

File: pkgs/misc/vscode-extensions/nix/default.nix [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  buildVscodeMarketplaceExtension {
+    mktplcRef = {
+        name = "Nix";
+        publisher = "bbenoist";
+        version = "1.0.1";
+        sha256 = "0zd0n9f5z1f0ckzfjr38xw2zzmcxg1gjrava7yahg5cvdcw6l35b";
+    };
+    meta = with stdenv.lib; {
+      description = ''
+        Pull Request Provider for GitHub
```
Comment (eadwu) [inline pkgs/misc/vscode-extensions/nix/default.nix]: Correct description?

**Comment by eadwu** [comment]
The update script currently in the repository works fine in terms of updating, I usually just copy over the updated versions and hashes when I spot an update when I run it.

**Comment by aanderse** [comment]
Is anyone interested in continuing with this PR?

**Comment by garbas** [comment]
@aanderse i just pushed the code where i stopped working on this. maybe it will inspire you to continue. i switched to back to neovim since coc.nvim plugin gives me everything that vscode does. i will close this PR but feel free to take my work and continue it.

**Comment by aanderse** [comment]
Thanks for the update @garbas!

**Pull request closed without merging**